crates/scarb-api/src/command.rs

```diff
@@ -0,0 +1,220 @@
+use anyhow::Context;
+use scarb_ui::args::PackagesFilter;
+use std::collections::HashMap;
+use std::ffi::{OsStr, OsString};
+use std::path::PathBuf;
+use std::process::{Command, Stdio};
+use std::{env, io};
+use thiserror::Error;
+
+/// Error thrown while trying to execute `scarb` command.
+#[derive(Error, Debug)]
+#[non_exhaustive]
+pub enum ScarbCommandError {
+    /// Failed to read `scarb` output.
+    #[error("failed to read `scarb` output")]
+    Io(#[from] io::Error),
+    /// Error during execution of `scarb` command.
+    #[error("`scarb` exited with error")]
+    ScarbError,
+}
+
+/// A builder for `scarb` command invocation.
+#[derive(Clone, Debug, Default)]
+#[allow(clippy::struct_excessive_bools)]
+pub struct ScarbCommand {
```

Comment:
I wonder if you don't want to have it as a part of scarb-ui or sth 🤔

Reply:
Nope - we have `ScarbCommand` in `scarb_metadata`, but it's intended for simple case when you inherit stdio and run command only.